api/src/test/java/edu/cornell/mannlib/vitro/webapp/dynapi/RESTEndpointTest.java

```diff
@@ -170,104 +167,28 @@ private void run(String method) {
-		prepareMocks(method, actionName);
-		when(httpMethod.getName()).thenReturn("FUBAR");
-	}
-
-	private void prepareMocksCustomAction(String method, String actionName) {
-		when(request.getMethod()).thenReturn(method);
-		when(request.getServletPath()).thenReturn(REST_SERVLET_PATH);
-		when(request.getPathInfo()).thenReturn(PATH_INFO + "/" + actionName);
-
-		when(action.run(any(OperationData.class)))
-			.thenReturn(new OperationResult(HttpServletResponse.SC_OK));
-
-		when(httpMethod.getName()).thenReturn(method);
-
-		when(rpc.getName()).thenReturn(actionName);
-		when(rpc.getHttpMethod()).thenReturn(httpMethod);
-
-		when(resource.getCustomRestActionRPC(actionName)).thenReturn(rpc);
-		doNothing().when(resource).removeClient();
-	}
-
-	private void prepareMocksUnsupportedCustomAction(String method, String actionName) {
-		prepareMocksCustomAction(method, actionName);
-		when(resource.getCustomRestActionRPC(actionName))
-			.thenThrow(new UnsupportedOperationException("Unsupported custom action"));
-	}
-
-	private void verifyMocksOk() {
-		verifyMocks(new int[] { 1, 0, 1, 1, 1, 1 }, HttpServletResponse.SC_OK);
-	}
-
-	private void verifyMocksNotFound() {
-		verifyMocks(new int[] { 0, 0, 0, 0, 0, 1 }, HttpServletResponse.SC_NOT_FOUND);
-	}
-
-	private void verifyMocksUnsupportedMethod() {
-		verifyMocks(new int[] { 1, 0, 1, 0, 0, 1 }, HttpServletResponse.SC_NOT_IMPLEMENTED);
-	}
-
-	private void verifyMocksRPCNotImplemented() {
-		verifyMocks(new int[] { 1, 0, 1, 0, 0, 1 }, HttpServletResponse.SC_NOT_IMPLEMENTED);
-	}
-
-	private void verifyMocksCustomActionOk() {
-		verifyMocks(new int[] { 0, 1, 1, 1, 1, 1 }, HttpServletResponse.SC_OK);
-	}
-
-	private void verifyMocksCustomActionUnsupported() {
-		verifyMocks(new int[] { 0, 1, 1, 0, 0, 1 }, HttpServletResponse.SC_NOT_IMPLEMENTED);
-	}
-
-	private void verifyMocks(int[] times, int status) {
-		verify(resource, times(times[0])).getRestRPC(any());
-		verify(resource, times(times[1])).getCustomRestActionRPC(any());
-		verify(resource, times(times[2])).removeClient();
-		verify(action, times(times[3])).run(any());
-		verify(action, times(times[4])).removeClient();
-		verify(response, times(times[5])).setStatus(status);
-	}
-
 	@Parameterized.Parameters
-	public static Collection<String[]> requests() {
-		return Arrays.asList(new String[][] {
-			// method action
-			{ "POST", "test" },
-			{ "GET", "test" },
-			{ "PUT", "test" },
-			{ "PATCH", "test" },
-			{ "DELETE", "test" }
+	public static Collection<Object[]> requests() {
+		String actionName = "test";
+		String customRestActionPathInfo = format("%s/%s", PATH_INFO, actionName);
+
+		return Arrays.asList(new Object[][] {
+			// expected counts key
+			// resource.getRestRPC, resource.getCustomRestActionRPC, resource.removeClient, action.run, action.removeClient, response.setStatus
+
+			// method   path info                 action name  expected counts                 expected status
```

Comment:
I overlooked this last time I reviewed.

Did you want to put a message on each of these, so that the failure reporting can be improved in the same way I did on the RPC tests?

Reply:
Yes, thanks.